matRad_calcCombiDose.m

```diff
@@ -53,23 +54,49 @@
         currPln = pln.originalPlans(k);
         currPln.bioParam = pln.originalPlans(k).bioParam;
 
-
         if strcmp(radiationModalities{k},'photons')
             dijt = [dijt, {matRad_calcPhotonDose(ct,currStf,currPln,cst,CalcDoseDirect)}];
+           
+            cst = matRad_setOverlapPriorities(cst,ct.cubeDim);
+            [ax,bx] = matRad_getPhotonLQMParameters(matRad_resizeCstToGrid(cst,dijt{k}.ctGrid.x,  dijt{k}.ctGrid.y,  dijt{k}.ctGrid.z,...
+                              dijt{k}.doseGrid.x,dijt{k}.doseGrid.y,dijt{k}.doseGrid.z),dijt{k}.doseGrid.numOfVoxels,1);
+
+            switch pln.bioParam.quantityOpt % later on, if change th effect/varRBE BP to avoid calculation of mAlphaDose/mSqrtBetaDose, just check if quantityopt is not pD, compute ax,bx,RBE and store them in dij
+                case {'physicalDose'}
+
+                case {'effect'}
+                    dijt{k}.ax = ax;
+                    dijt{k}.bx = bx;
+                    dijt{k}.abx = ax./bx;
+                    dijt{k}.mAlphaDose{1} = dijt{k}.physicalDose{1}.*ax;
+                    dijt{k}.mSqrtBetaDose{1} = dijt{k}.physicalDose{1}.*sqrt(bx);
                 
-            if ~strcmp(pln.originalPlans(k).bioParam.quantityOpt, 'physicalDose')
-                cst = matRad_setOverlapPriorities(cst,ct.cubeDim);
-                [ax,bx] = matRad_getPhotonLQMParameters(matRad_resizeCstToGrid(cst,dijt{k}.ctGrid.x,  dijt{k}.ctGrid.y,  dijt{k}.ctGrid.z,...
-                                 dijt{k}.doseGrid.x,dijt{k}.doseGrid.y,dijt{k}.doseGrid.z),dijt{k}.doseGrid.numOfVoxels,1);
-                dijt{k}.ax = ax;
-                dijt{k}.bx = bx;
-                dijt{k}.abx = ax./bx;
-                dijt{k}.mAlphaDose{1} = dijt{k}.physicalDose{1}.*ax;
-                dijt{k}.mSqrtBetaDose{1} = dijt{k}.physicalDose{1}.*sqrt(bx);
+                case {'RBExD'}
+
+                    
+                    % if any(strcmp({bioModels.model}, 'constRBE'))
+                    %     dijt{k}.RBE = 1;
+                    % else
+                        dijt{k}.RBE = 1;
+                        dijt{k}.ax = ax;
+                        dijt{k}.bx = bx;
+                        dijt{k}.abx = ax./bx;
+                        dijt{k}.mAlphaDose{1} = dijt{k}.physicalDose{1}.*ax;
+                        dijt{k}.mSqrtBetaDose{1} = dijt{k}.physicalDose{1}.*sqrt(bx);
+
+                        dijt{k}.ixDose  = dijt{k}.bx~=0;
+                        dijt{k}.gamma   = zeros(dijt{k}.doseGrid.numOfVoxels,1);
+                        dijt{k}.gamma(dijt{k}.ixDose) = dijt{k}.ax(dijt{k}.ixDose)./(2*dijt{k}.bx(dijt{k}.ixDose));
+                        
+
+                    %end
+            
             end
 
         elseif strcmp(radiationModalities{k},'protons') || strcmp(radiationModalities{k},'carbon')|| strcmp(radiationModalities{k},'helium')
             dijt = [dijt, {matRad_calcParticleDose(ct,currStf,currPln,cst,CalcDoseDirect)}];
+
```

Comment:
What about proton const RBE effect optimization, needs to have a block similar to the photon effect  parameters here